src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -2799,6 +2907,20 @@ public override SyntaxNode WithStatements(SyntaxNode declaration, IEnumerable<Sy
             }
         }
 
+        private SyntaxNode FixDeclarationWithStatements(SyntaxNode declaration)
+        {
+            if (HasBody(declaration))
+            {
+                var mods = GetModifiers(declaration);
+                if (mods.IsAbstract)
+                {
+                    return WithModifiers(declaration, mods.WithIsAbstract(false).WithIsVirtual(true));
```

Comment:
I'm concerned about us randomly adding modifiers that weren't asked for.  What if i want an abstract method with a body?  That choice will just be overridden?